core/ecschema-metadata/src/SchemaFormatsProvider.ts

```diff
@@ -7,125 +7,121 @@
  */
 
 import { ISchemaLocater, SchemaContext } from "./Context";
-import { FormatsProvider } from "./Interfaces";
 import { SchemaItemKey, SchemaKey } from "./SchemaKey";
 import { SchemaItem } from "./Metadata/SchemaItem";
 import { Format } from "./Metadata/Format";
 import { SchemaItemFormatProps } from "./Deserialization/JsonProps";
-import { BeUiEvent } from "@itwin/core-bentley";
+import { BeEvent } from "@itwin/core-bentley";
 import { KindOfQuantity } from "./Metadata/KindOfQuantity";
 import { getFormatProps } from "./Metadata/OverrideFormat";
-import { UnitSystemKey } from "@itwin/core-quantity";
+import { FormatProps, FormatsChangedArgs, FormatsProvider, UnitSystemKey } from "@itwin/core-quantity";
+import { Unit } from "./Metadata/Unit";
+import { InvertedUnit } from "./Metadata/InvertedUnit";
 
 /**
  * Provides default formats and kind of quantities from a given SchemaContext or SchemaLocater.
  * @beta
  */
 export class SchemaFormatsProvider implements FormatsProvider {
   private _context: SchemaContext;
-  private _formatCache: Map<string, SchemaItemFormatProps> = new Map();
-  public onFormatUpdated = new BeUiEvent<string>();
-  public onFormatRemoved = new BeUiEvent<string>();
-  public onCacheCleared = new BeUiEvent<void>();
+  private _unitSystem: UnitSystemKey;
+  private _formatsRetrieved: Set<string> = new Set();
+  public onFormatsChanged = new BeEvent<(args: FormatsChangedArgs) => void>();
   /**
    *
    * @param contextOrLocater The SchemaContext or a different ISchemaLocater implementation used to retrieve the schema. The SchemaContext
    * class implements the ISchemaLocater interface. If the provided locater is not a SchemaContext instance a new SchemaContext will be
    * created and the locater will be added.
-   * @param _unitExtraData Additional data like alternate display label not found in Units Schema to match with Units; Defaults to empty array.
```

Comment:
add doc for unitSystem parameter

Reply:
OK, I see where you describe it.  Not sure if that is the best place but maybe improving it is best to a second pass that will work on documentation

Reply:
Done, I added a @param field for unitSystem.